backend/test/mission_service_impl_test.cpp

```diff
@@ -386,6 +389,53 @@ TEST_F(MissionServiceImplIsFinishedTest, isMissionFinishedDoesNotCrashWithNullRe
     _mission_service.IsMissionFinished(nullptr, nullptr, nullptr);
 }
 
+class MissionServiceImplPauseTest : public MissionServiceImplTestBase
+{
+protected:
+    std::future<void> pauseMissionAndSaveParams(std::shared_ptr<PauseMissionResponse> response);
+};
+
+INSTANTIATE_TEST_CASE_P(MissionResultCorrespondences,
+                        MissionServiceImplPauseTest,
+                        ::testing::ValuesIn(generateInputPairs()));
+
+ACTION_P2(SaveResult, callback, callback_saved_promise)
```

Comment:
This `ACTION_P2` is already defined above. I believe you don't need to redefine it here. Can you test without it? :)

Reply:
okay sure.